crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -319,17 +334,16 @@ impl<'a> Submitter<'a> {
                 return SubmissionError::from(err);
             }
 
-            // if gas price has increased cancel old transaction so new transaction could be submitted.
-
-            if let Some((previous_tx, previous_gas_price)) = transactions.last() {
+            // if gas price has not increased enough, skip submitting the transaction.
+            if let Some(previous_gas_price) = transactions
+                .last()
+                .map(|(_, previous_gas_price)| previous_gas_price)
+                .or_else(|| pending_gas_price.as_ref())
+            {
                 let previous_gas_price = previous_gas_price.bump(1.125).ceil();
-                if gas_price.cap() > previous_gas_price.cap()
-                    && gas_price.tip() > previous_gas_price.tip()
+                if gas_price.tip() < previous_gas_price.tip()
+                    || gas_price.cap() < previous_gas_price.cap()
```

Comment:
I think this is a subtle change in behavior. Shouldn't it be `gas_price.tip() <= previous_gas_price.tip()` and same for `.cap()` to invert the if?

Also I'm not sure about how the gas price has to change in order to warrant resubmitting the order with a higher gas price. Do the `.tip()` *AND* `.cap()` of the current gas price need to be higher than what we submitted previously? Intuitively I would think it's enough if either value exceeds what we previously submitted.